docs/content/latest/explore/ysql-language-features/data-manipulation.md

```diff
@@ -0,0 +1,343 @@
+
+The `COPY TO` statement allows you to export data from a table to a file. By specifying a column list, you can instruct `COPY TO` to only export data from certain columns.
+
+The `filename` parameter of the `COPY` statement enables copying to a file directly.
+
+The following example demonstrates how to use the `COPY FROM` statement:
+
+```sql
+COPY employees TO '/home/mydir/employees.txt.sql' DELIMITER ',';
+```
+
+## Backing Up a Database
+
+You can back up a single instance of a YugabyteDB database into a plain-text SQL file by using the  `ysql_dump` is a utility, as follows:
+
+```
+ysql_dump mydb > mydb.sql
+```
+
+To back up global objects that are common to all databases in a cluster, such as roles, you need to use  `ysql_dumpall` .
+
+`ysql_dump` makes backups regardless of whether or not the database is being used.
+
+To reconstruct the database from a plain-text SQL file to the state the database was in at the time of saving, import this file using the `ysqlsh \i`  command, as follows:
+
+```
+ysqlsh \i mydb
+```
+
+## Configuring Automatic Timestamps
+
+You can use automatic timestamps to keep track of when data in a table was added or updated. 
+
+The date of the data creation is typically added via a  `created_at` column with a default value of `NOW()`, as shown in the following example: 
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text,
+    department text,
+  	created_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+```
+
+To track updates, you need to use triggers that let you define functions executed when an update is performed. The following example shows how to create a function in PL/pgSQL that returns an object called `NEW` containing data being modified:
+
+```sql
+CREATE OR REPLACE FUNCTION trigger_timestamp()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+The following examples create a table and connect it with a trigger that executes the `trigger_timestamp` function every time a row is updated in the table:
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text,
+    department text,
+  	updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+)
+```
+
+```sql
+CREATE TRIGGER set_timestamp
+BEFORE UPDATE ON employees
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_timestamp();
+```
+
+## Defining NOT NULL Constraint
+
+YSQL lets you define various constraints on columns. One of these constrains is `NOT NULL`. You can use it to specify that a column value cannot be null. Typically, you apply this constraint when creating a table, as the following example demonstrates:
+
+```sql
+CREATE TABLE employees (
+    employee_no integer NOT NULL,
+    name text NOT NULL,
+    department text
+);
+```
+
+You may also add the `NOT NULL` constraint to one or more columns of an existing table by executing the `ALTER TABLE` statement, as follows:
+
+```sql
+ALTER TABLE employees
+ALTER COLUMN department SET NOT NULL;
+```
+
```

Comment:
I would suggest demonstrating the insert that does not include the no `NOT NULL` column, to demonstrate the protection.

Reply:
Thanks